### PR TITLE
Add DesignerTransaction tests and fix debug assertions

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerTransaction.cs [hunk 1]
```diff
@@ -81,51 +53,36 @@ public void Commit()
...
+        /// Private implementation of IDisaposable. When a transaction is disposed
+        /// it is committed.
         /// </summary>
         void IDisposable.Dispose()
         {
             Dispose(true);
 
-            // note - Dispose calls Cancel which sets this bit, so
-            //        this should never be hit.
-            //
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerTransaction.cs]: Why was this and the assert removed?

File: src/System.ComponentModel.TypeConverter/tests/Design/DesignerTransactionTests.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+using Xunit;
+
+namespace System.ComponentModel.Design.Tests
+{
+    public class DesignerTransactionTests
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            var transaction = new TestDesignerTransaction();
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/tests/Design/DesignerTransactionTests.cs]: All of these instances should be disposed of.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerTransaction.cs (hunk 1 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesignerTransaction.cs]: Dispose is virtual, the second test here triggers the Debug.Fail

```cs
[Fact]
public void Dispose_FinalizeSuppressed_Success()
{
    var transaction = new NonDisposingDesignerTransaction();
    transaction.Cancel();
    ((IDisposable)transaction).Dispose();

    Assert.True(transaction.Canceled);
}

[Fact]
public void Dispose_FinalizeNotSuppressed_Success()
{
    var transaction = new NonDisposingDesignerTransaction();
    ((IDisposable)transaction).Dispose();

    Assert.False(transaction.Canceled);
}

private class NonDisposingDesignerTransaction : DesignerTransaction
{
    protected override void Dispose(bool disposing) { }

    protected override void OnCancel() { }
    protected override void OnCommit() { }
}
```

**Comment by safern** [comment]
test OSX x64 Debug Build please
test Windows x64 Debug Build please

**Pull request merged**